src/devices/cpu/tlcs900/tmp94c241.cpp

```diff
@@ -0,0 +1,1935 @@
+	map(0x0000f7, 0x0000f7).rw(FUNC(tmp94c241_device::intnmwdt_r), FUNC(tmp94c241_device::intnmwdt_w));
+	map(0x0000f8, 0x0000f8).w(FUNC(tmp94c241_device::intclr_w));
+	map(0x000100, 0x000103).w(FUNC(tmp94c241_device::dmav_w));
+	map(0x000110, 0x000110).rw(FUNC(tmp94c241_device::wdmod_r), FUNC(tmp94c241_device::wdmod_w));
+	map(0x000111, 0x000111).w(FUNC(tmp94c241_device::wdcr_w));
+	map(0x000120, 0x000127).r(FUNC(tmp94c241_device::adreg_r));
+	map(0x000128, 0x000128).rw(FUNC(tmp94c241_device::admod1_r), FUNC(tmp94c241_device::admod1_w));
+	map(0x000129, 0x000129).rw(FUNC(tmp94c241_device::admod2_r), FUNC(tmp94c241_device::admod2_w));
+	map(0x000130, 0x000131).w(FUNC(tmp94c241_device::dareg_w));
+	map(0x000132, 0x000132).rw(FUNC(tmp94c241_device::dadrv_r), FUNC(tmp94c241_device::dadrv_w));
+	map(0x000140, 0x000141).w(FUNC(tmp94c241_device::b0cs_w));
+	map(0x000142, 0x000142).rw(FUNC(tmp94c241_device::mamr0_r), FUNC(tmp94c241_device::mamr0_w));
+	map(0x000143, 0x000143).rw(FUNC(tmp94c241_device::msar0_r), FUNC(tmp94c241_device::msar0_w));
+	map(0x000144, 0x000145).w(FUNC(tmp94c241_device::b1cs_w));
+	map(0x000146, 0x000146).rw(FUNC(tmp94c241_device::mamr1_r), FUNC(tmp94c241_device::mamr1_w));
+	map(0x000147, 0x000147).rw(FUNC(tmp94c241_device::msar1_r), FUNC(tmp94c241_device::msar1_w));
+	map(0x000148, 0x000149).w(FUNC(tmp94c241_device::b2cs_w));
+	map(0x00014a, 0x00014a).rw(FUNC(tmp94c241_device::mamr2_r), FUNC(tmp94c241_device::mamr2_w));
+	map(0x00014b, 0x00014b).rw(FUNC(tmp94c241_device::msar2_r), FUNC(tmp94c241_device::msar2_w));
+	map(0x00014c, 0x00014d).w(FUNC(tmp94c241_device::b3cs_w));
+	map(0x00014e, 0x00014e).rw(FUNC(tmp94c241_device::mamr3_r), FUNC(tmp94c241_device::mamr3_w));
+	map(0x00014f, 0x00014f).rw(FUNC(tmp94c241_device::msar3_r), FUNC(tmp94c241_device::msar3_w));
+	map(0x000150, 0x000151).w(FUNC(tmp94c241_device::b4cs_w));
+	map(0x000152, 0x000152).rw(FUNC(tmp94c241_device::mamr4_r), FUNC(tmp94c241_device::mamr4_w));
+	map(0x000153, 0x000153).rw(FUNC(tmp94c241_device::msar4_r), FUNC(tmp94c241_device::msar4_w));
+	map(0x000154, 0x000155).w(FUNC(tmp94c241_device::b5cs_w));
+	map(0x000156, 0x000156).rw(FUNC(tmp94c241_device::mamr5_r), FUNC(tmp94c241_device::mamr5_w));
+	map(0x000157, 0x000157).rw(FUNC(tmp94c241_device::msar5_r), FUNC(tmp94c241_device::msar5_w));
+	map(0x000400, 0x000bff).ram();
+}
+
+//**************************************************************************
+//  EXECUTION CALLBACKS
+//**************************************************************************
+
+//-------------------------------------------------
+//  tlcs900_check_hdma -
+//-------------------------------------------------
+
+void tmp94c241_device::tlcs900_check_hdma()
+{
+}
+
+
+//-------------------------------------------------
+//  tlcs900_check_irqs -
+//-------------------------------------------------
+
+void tmp94c241_device::tlcs900_check_irqs()
+{
+	int irq_vectors[8] = { -1, -1, -1, -1, -1, -1, -1, -1 };
```

Comment:
Please put variable declarations closer to their first use, C99 is 26 years old at this point and C++ had them even earlier :-)

The int i is particularly egregious, an index variable should escape its loop if and only if its value is used after the loop.

Reply:
Fixed this now.